packages/rpc/src/client.ts

```diff
@@ -108,173 +137,183 @@ export class RPCClient extends EventEmitter {
 
 		/**
 		 * Application used in this client
-		 * @type {?ClientApplication}
 		 */
 		this.application = null;
 
 		/**
 		 * User used in this application
-		 * @type {?Partial<APIUser>}
 		 */
 		this.user = null;
 
-		const Transport = options.transport ? transports[options.transport] : false;
-		if (!Transport) {
-			throw new TypeError('RPC_INVALID_TRANSPORT', options.transport as undefined);
+		const transport = options.transport ? transports[options.transport] : false;
+		if (!transport) {
+			const typeError = new TypeError('INVALID_TRANSPORT', options.transport as undefined);
+			Error.captureStackTrace(typeError, this.constructor);
+			throw typeError;
 		}
 
-		this.fetch = (method, path, options) =>
-			fetch(`${this.endpoint}${path}${options?.query ? new URLSearchParams(options.query).toString() : ''}`, {
-				method,
-				body: options?.data.toString() as string | null,
-				headers: {
-					Authorization: `Bearer ${this.accessToken!}`,
+		this.fetch = async (method, path, options) => {
+			const res = await fetch(
+				`${this.endpoint}${path}${options?.query ? new URLSearchParams(options.query).toString() : ''}`,
+				{
+					method,
+					body: options?.data.toString() as string | null,
+					headers: {
+						Authorization: `Bearer ${this.accessToken!}`,
+					},
 				},
```

Comment:
Should `@discordjs/rest` be used here instead?

Reply:
working on it

Reply:
actually, porting the function to @discordjs/rest can be done by other contributor, i have no idea how @discordjs/rest work nor how error handling work